File: quickparcel/quickparcel/src/main/java/edu/cit/lariosa/quickparcel/entity/Payment.java
package edu.cit.lariosa.quickparcel.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Table(name = "payments")
@Data
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "delivery_id", unique = true, nullable = false)
    private Delivery delivery;

    @ManyToOne
    @JoinColumn(name = "sender_id", nullable = false)
    private Sender sender;

    @Column(name = "payment_reference")
    private String paymentReference;

    @Column(name = "transaction_id")
    private String transactionId;

    @Column(nullable = false)
    private Double amount;

    private String currency = "PHP";

    @Column(nullable = false)
    private String status; // PENDING, COMPLETED, FAILED, REFUNDED

    @Column(name = "payment_method")
    private String paymentMethod;

    @Column(name = "payer_email")
    private String payerEmail;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "completed_at")
    private LocalDateTime completedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: frontend/src/services/payment.js
import api from './api';

export const createPayMongoPayment = async (deliveryId, amount, description) => {
  try {
    const response = await api.post('/payments/paymongo/create', {
      deliveryId,
      amount,
      description
    });
    return response.data;
  } catch (error) {
    console.error('PayMongo payment creation failed:', error);
    throw error;
  }
};

export const getPayMongoPaymentStatus = async (paymentReference) => {
  try {
    const response = await api.get(`/payments/paymongo/status/${paymentReference}`);
    return response.data;
  } catch (error) {
    console.error('Failed to get payment status:', error);
    throw error;
  }
};